substrafl/remote/register/register.py

```diff
@@ -29,21 +29,39 @@
 MINIMAL_PYTHON_VERSION = 9  # 3.9
 MAXIMAL_PYTHON_VERSION = 12  # 3.12
 
-_DEFAULT_BASE_DOCKER_IMAGE = "python:{python_version}-slim"
+_CPU_BASE_IMAGE = """
+FROM python:{python_version}-slim
 
-DOCKERFILE_TEMPLATE = """
-FROM {docker_image}
+# update image
+RUN apt-get update -y
+"""
+
+_GPU_BASE_IMAGE = """
+FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04
 
 # update image
-RUN apt update -y
+RUN apt-get update -y
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get install -y software-properties-common
```

Comment:
I would recommend do that in one layer and also remove the cache (cf https://docs.docker.com/build/building/best-practices/#apt-get)